engine/apps/api/views/schedule.py

```diff
@@ -154,6 +155,7 @@ def _annotate_queryset(self, queryset):
     def get_queryset(self):
         is_short_request = self.request.query_params.get("short", "false") == "true"
         filter_by_type = self.request.query_params.get("type")
+        used = BooleanField(allow_null=True).to_internal_value(data=self.request.query_params.get("used"))
```

Comment:
Maybe we can use something easier, like
` is_short_request = self.request.query_params.get("short", "false") == "true"`
( but with check is filter is present)

Reply:
I like `BooleanField` here because it automatically handles null values as well as the case when the param is not provided. It also returns 400 in case non-boolean value is provided (e.g. 400 will be returned on `?used=random`)